packages/react-data-grid/src/Cell.tsx

```diff
@@ -233,16 +169,38 @@ export default class Cell extends React.PureComponent {
       onDragOver: this.handleDragOver
     };
 
-    if (!columnEvents || !onColumnEvent) {
-      return gridEvents;
+    if (!columnEvents) {
+      return allEvents;
+    }
+
+    for (const event in columnEvents) {
+      if (columnEvents.hasOwnProperty(event)) {
+        const eventInfo: ColumnEventInfo = {
+          idx,
+          rowIdx,
+          column,
+          rowId: rowData[cellMetaData.rowKey]
+        };
+        if (allEvents.hasOwnProperty(event)) {
+          const existingEvent = allEvents[event];
+          allEvents[event] = (e: Event) => {
+            existingEvent(e);
+            columnEvents[event](e, eventInfo);
+          };
+        } else {
+          allEvents[event] = (e: Event) => {
+            columnEvents[event](e, eventInfo);
+          };
+        }
+      }
     }
 
-    return this.createEventDTO(gridEvents, columnEvents, onColumnEvent);
```

Comment:
I think the whole purpose of this code is to add an event it does not exist or call the old event along with new if it exists. This can be really simplified